lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
+    values = [values];
+  }
+
+  var valuesIndex = 0;
+  var chunkIndex = placeholdersRegex.lastIndex = 0;
```

Comment:
So after looking this over and playing around, I realize hat we need to make sure that `SqlString.format` is re-entrant, since we are calling out to non-private functions (they are properties on the `SqlString` object that can technically be overwritten) within the `while` loop. How much of a performance hit would it be to construct the `placeholdersRegex` on each `.format()` call?